k3s-io/k3s
# Auto import images for containerd image store

#### Proposed Changes ####



- ADR
- Auto import images form .txt files and tar files

#### Types of Changes ####



- New Feature

#### Verification ####



- run a k3s server
```shell
k3s server
```
- create the images folder
```shell
mkdir /var/lib/rancher/k3s/agent/images
```

- create a .txt file and add a docker.io image path to the .txt file
```shell
docker.io/library/mysql:latest
```

- move the file to /var/lib/rancher/k3s/agent/images
```shell
cp example.txt /var/lib/rancher/k3s/agent/images
```

- then see if the file was imported in the containerd image store
```shell
k3s ctr images list | grep "mysql"
```

- you can also change the file inside var/lib/rancher/k3s/agent/images
```shell
nano /var/lib/rancher/k3s/agent/images/example.txt
```

- then you can add 
```shell
docker.io/library/redis:latest
```

- the controller will see the change and then import the redis image
```shell
k3s ctr images list | grep "redis"
```

#### Testing ####




#### Linked Issues ####



- https://github.com/k3s-io/k3s/issues/9759
- https://github.com/k3s-io/k3s/issues/8084
#### User-Facing Change ####

```release-note
Users can now auto import images to containerd by just throwing the image in the agent/images folder while k3s is running
```

#### Further Comments ####
